src/services/resources-getter.js

```diff
@@ -1,255 +1,78 @@
+    return model.count({
+      ...options,
+      col: _.isEmpty(this._model.primaryKeys) ? '*' : undefined,
     });
-
-    return hasCustomFieldSearch;
-  }
-
-  async getRecords(searchBuilder, fieldNamesRequested) {
-    const segment = await this.getSegment();
-    const segmentScope = segment && segment.scope;
-    const scope = segmentScope ? this.model.scope(segmentScope) : this.model.unscoped();
-    const include = this.queryBuilder.getIncludes(this.model, fieldNamesRequested);
-
-    const where = await this.buildWhereConditions(searchBuilder, this.params, segment);
-
-    const queryOptions = {
-      where,
-      include,
-      order: this.queryBuilder.getOrder(),
-      offset: this.queryBuilder.getSkip(),
-      limit: this.queryBuilder.getLimit(),
-    };
-
-    const { search, searchExtended } = this.params;
-
-    if (search) {
-      const hasCustomFieldSearch = this.addCustomFieldSearchToQueryOptions(
-        queryOptions, this.schema.fields, search,
-      );
-
-      const fieldsSearched = searchBuilder.getFieldsSearched();
-
-      if (fieldsSearched.length === 0 && !hasCustomFieldSearch) {
-        if (!searchExtended || !searchBuilder.hasExtendedSearchConditions()) {
-          // NOTICE: No search condition has been set for the current search, no record can be
-          //         found.
-          return [];
-        }
-      }
-    }
-
-    return scope.findAll(queryOptions);
   }
 
-
-  async count() {
-    const fieldNamesRequested = await this.getFieldNamesRequested();
-    const searchBuilder = new SearchBuilder(
-      this.model,
-      this.options,
-      this.params,
-      fieldNamesRequested,
-    );
-    const segment = await this.getSegment();
-    const segmentScope = segment && segment.scope;
-    const scope = segmentScope ? this.model.scope(segmentScope) : this.model.unscoped();
-    const include = this.queryBuilder.getIncludes(this.model, fieldNamesRequested);
-
-    const where = await this.buildWhereConditions(searchBuilder, this.params, segment);
-
-    const queryOptions = {
-      include,
-      where,
-    };
-
-    if (!this.primaryKey) {
-      // NOTICE: If no primary key is found, use * as a fallback for Sequelize.
-      queryOptions.col = '*';
-    }
-
-    const { search, searchExtended } = this.params;
-
-    if (search) {
-      const hasCustomFieldSearch = this.addCustomFieldSearchToQueryOptions(
-        queryOptions, this.schema.fields, search,
-      );
-
-      const fieldsSearched = searchBuilder.getFieldsSearched();
-
-      if (fieldsSearched.length === 0 && !hasCustomFieldSearch) {
-        if (!searchExtended
-          || !searchBuilder.hasExtendedSearchConditions()) {
-          // NOTICE: No search condition has been set for the current search, no record can be
-          //         found.
-          return 0;
-        }
-      }
-    }
-
-    return scope.count(queryOptions);
+  /** Load records matching current query (w/ pagination) */
+  async _getRecords() {
```

Comment:
AVoid reordering method on refacto, it increase the diff and it's way harder for the reviewer to see the difference

Reply:
There is no difference in this file.
I deleted the old one and rewrote it from scratch...